File: VB/Demos/08FunctionalProgramming/Person.vb
Namespace FunctionalProgramming
  Class Person
    Implements IComparable(Of Person)

    Property Id As Guid
    Property FirstName As String
    Property LastName As String


    Public Function CompareTo(other As Person) As Integer Implements IComparable(Of Person).CompareTo
      Return FirstName.CompareTo(other.FirstName)
    End Function
  End Class
End Namespace

File: CS/Demos/08FunctionalProgramming/Person.cs
namespace Demos.FunctionalProgramming;

class Person(Guid id, string firstName, string lastName) : IComparable<Person>
{
  public Guid Id { get; set; } = id;
  public string FirstName { get; set; } = firstName;
  public string LastName { get; set; } = lastName;

  public int CompareTo(Person? other)
  {
    return FirstName.CompareTo(other?.FirstName);
  }
}
